### PR TITLE
feat: zapper validators revamp

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- Refactors the myzod schema for the Zapper endpoints to leverage the existing ones, adding a new `V2AppsBalancesResponse` schema. Note, many fields have been set to optional. The reason for that is that while the current schema worked for the current endpoints, the same schemas (e.g `ZapperAsset`, `ZapperToken`) end up having a lot of fields maybe undefined in `/v2/balances/apps`
- ~~Adds RTK queries to fetch read-only opportunities and massage them as minimal data. Note that in addition to the zapperApi, and `opportunitiesSlice` query/resolver pair has been added, effectively just a wrapper on top of the zapper RTK query. This is because of circular dependencies issues and will ensure we don't get any.~~

Note, this is *not* consumed yet and will be wired up in a follow-up PR.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A - https://www.notion.so/shapeshift/Displaying-Earn-DeFi-positions-5fbfca607cd3491cb621e87242328209

## Risk

Low and realistically close to none. As a paranoia risk, the refactor in the current schema might break current UNI-V2 opportunities, test accordingly

## Testing

- Clear your store and ensure UNI-V2 opportunities / Zapper NFTs are fetched

### Engineering

- ☝🏽 

### Operations

- ☝🏽

### CODE REVIEW

File: src/state/apis/zapper/zapperApi.ts [hunk 1]
```diff
@@ -254,4 +337,8 @@ export const zapper = createApi({
   }),
 })
 
-export const { useGetZapperNftUserTokensQuery, useGetZapperCollectionsQuery } = zapperApi
+export const {
+  useGetZapperAppsbalancesOutputQuery,
```
Comment (gomesalexandre) [inline src/state/apis/zapper/zapperApi.ts:341]:
```suggestion
  useGetZapperAppsBalancesOutputQuery,
```

File: src/state/apis/zapper/zapperApi.ts [hunk 2]
```diff
@@ -131,18 +132,18 @@ export const zapperApi = createApi({
...
 
             acc.byId[assetId] = makeAsset({
               assetId,
-              symbol: appTokenData.symbol,
+              symbol: appTokenData.symbol ?? '',
               // WETH should be displayed as ETH in the UI due to the way UNI-V2 works
               // ETH is used for depositing/withdrawing, but WETH is used under the hood
               name,
-              precision: appTokenData.decimals,
+              precision: appTokenData.decimals ?? 0,
```
Comment (0xdef1cafe) [inline src/state/apis/zapper/zapperApi.ts:146]: if we don't have precision log and skip the asset, we never want to guess on this

File: src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts [hunk 3]
```diff
@@ -123,13 +121,13 @@ export const uniV2LpOpportunitiesMetadataResolver = async ({
...
       }
 
-      token0Decimals = zapperAppBalanceData.tokens[0].decimals
-      token1Decimals = zapperAppBalanceData.tokens[1].decimals
-      token0Reserves = bnOrZero(zapperAppBalanceData.dataProps.reserves[0])
-      token1Reserves = bnOrZero(zapperAppBalanceData.dataProps.reserves[1])
-      token0Address = ethers.utils.getAddress(zapperAppBalanceData.tokens[0].address)
-      token1Address = ethers.utils.getAddress(zapperAppBalanceData.tokens[1].address)
-      apr = bnOrZero(zapperAppBalanceData.dataProps.apy).toFixed()
+      token0Decimals = zapperAppBalanceData?.tokens?.[0].decimals!
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts:124]: feel like this will blow up on us

Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts]: 9c3049f80a0c87e4de5094e359e857d412a66e11

File: src/state/apis/zapper/zapperApi.ts (hunk 2 above)
Comment (gomesalexandre) [reply src/state/apis/zapper/zapperApi.ts]: 9c3049f80a0c87e4de5094e359e857d412a66e11

File: src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts (hunk 3 above)
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts]: Note we still need the non-null assertion because ts is a bit derpy here

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**